src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityFissionReactor.java

```diff
@@ -0,0 +1,1104 @@
+        super.receiveCustomData(dataId, buf);
+
+        if (dataId == GregtechDataCodes.SYNC_REACTOR_STATS) {
+            this.temperature = buf.readDouble();
+            this.maxTemperature = buf.readDouble();
+            this.pressure = buf.readDouble();
+            this.maxPressure = buf.readDouble();
+            this.power = buf.readDouble();
+            this.maxPower = buf.readDouble();
+            this.kEff = buf.readDouble();
+            this.controlRodInsertionValue = buf.readDouble();
+            this.fuelDepletionPercent = buf.readDouble();
+        } else if (dataId == GregtechDataCodes.SYNC_LOCKING_STATE) {
+            this.lockingState = buf.readEnumValue(LockingState.class);
+            this.scheduleRenderUpdate();
+        }
+    }
+
+    protected void lockAll() {
+        for (ICoolantHandler handler : this.getAbilities(MultiblockAbility.IMPORT_COOLANT)) {
+            handler.setLock(true);
+        }
+        for (IFuelRodHandler handler : this.getAbilities(MultiblockAbility.IMPORT_FUEL_ROD)) {
+            handler.setLock(true);
+        }
+    }
+
+    protected void unlockAll() {
+        for (ICoolantHandler handler : this.getAbilities(MultiblockAbility.IMPORT_COOLANT)) {
+            handler.setLock(false);
+        }
+        for (IFuelRodHandler handler : this.getAbilities(MultiblockAbility.IMPORT_FUEL_ROD)) {
+            handler.setLock(false);
+        }
+        if (this.fissionReactor != null) {
+            this.fissionReactor.turnOff();
+        }
+        if (this.lockingState == LockingState.LOCKED) { // Don't remove warnings
+            this.setLockingState(LockingState.UNLOCKED);
+        }
+    }
+
+    private void lockAndPrepareReactor() {
+        this.lockAll();
+        int radius = this.diameter / 2;     // This is the floor of the radius, the actual radius is 0.5 blocks
+        // larger
+        BlockPos reactorOrigin = this.getPos().offset(this.frontFacing.getOpposite(), radius);
+        boolean foundFuel = false;
+        for (int i = -radius; i <= radius; i++) {
+            for (int j = -radius; j <= radius; j++) {
+                if (Math.pow(i, 2) + Math.pow(j, 2) > Math.pow(radius, 2) + radius)         // (radius + .5)^2 =
+                    // radius^2 + radius + .25
+                    continue;
+                BlockPos currentPos = reactorOrigin.offset(this.getRight(), i)
+                        .offset(this.frontFacing.getOpposite(), j).offset(getUp(), heightTop);
+                if (getWorld().getTileEntity(currentPos) instanceof IGregTechTileEntity gtTe) {
+                    MetaTileEntity mte = gtTe.getMetaTileEntity();
+                    if (mte instanceof ICoolantHandler coolantIn) {
+                        Fluid lockedFluid = coolantIn.getLockedObject();
+                        if (lockedFluid != null) {
+                            Material mat = FluidUnifier.getMaterialFromFluid(lockedFluid);
```

Comment:
This should be **heavily** avoided. Please create a separate registry to store fluid name -> coolant data.